src/game/eop.js

```diff
@@ -116,321 +109,324 @@ export const ElevationOfPrivilege = Game({
-          let dealt = [...G.dealt];
-          let order = [...G.order];
-          let suit = G.suit;
-          let dealtBy = G.dealtBy;
-          let scores = [...G.scores];
-          let round = G.round;
-          let lastWinner = G.lastWinner;
-
-          // calculate the scores
-          if (dealt.length >= ctx.numPlayers) {
-            let idx = getWinner(suit, dealt);
-            lastWinner = order[idx];
-    
-            scores[lastWinner]++;
-    
-            order = [];
-            for (let i=0; i<ctx.numPlayers; i++) {
-              order.push(i);
-            }
-            for (let i=0; i<lastWinner; i++) {
-              let element = order.shift();
-              order.push(element);
-            }
-    
-            dealt = [];
-            suit = "";
-            dealtBy = "";
-            round++;
-          }
-
-          return {
-            ...G,
-            dealt,
-            lastWinner,
-            dealtBy,
-            order,
-            suit,
-            scores,
-            round,
-            passed: [], // reset the passed array when the phase ends
-          }
-        },
-      },
-      play: {
-        turnOrder: {
-          playOrder: function playOrder(G) {
-            return G.order;
-          },
-          first: function first(G, ctx) {
-            return G.playOrderPos % ctx.playOrder.length;
-          },
-          next: function next(G, ctx) {
-            return (ctx.playOrderPos + 1) % ctx.playOrder.length;
-          }
-        },
-        next: 'threats',
-        allowedMoves: ['draw', 'selectDiagram', 'selectComponent', 'selectThreat'],
-        onTurnEnd: (G, ctx) => {
-          ctx.events.endPhase();
-          return {
-            ...G,
-            playOrderPos: G.playOrderPos + 1,
-          }
-        },
-      },
+  // only update score if it's a new threat
+  if (G.threat.new) {
+    scores[ctx.playerID]++;
+  }
+
+  // TODO: have a cleaner or readable approach to updating this object
+  let identifiedThreats = _.cloneDeep(G.identifiedThreats);
+  
+  if (!(G.selectedDiagram in identifiedThreats)) {
+    Object.assign(identifiedThreats, {[G.selectedDiagram]: {}});
+  }
+
+  if (!(G.selectedComponent in identifiedThreats[G.selectedDiagram])) {
+    Object.assign(identifiedThreats[G.selectedDiagram], {[G.selectedComponent]: {}});
+  }
+
+  Object.assign(identifiedThreats[G.selectedDiagram][G.selectedComponent], {
+    [G.threat.id]: {
+      id: G.threat.id,
+      owner: G.threat.owner,
+      title: threat_title,
+      type: G.threat.type,
+      severity: G.threat.severity,
+      description: threat_description,
+      mitigation: threat_mitigation || "No mitigation provided.",
+    }
+  });
+
+  return {
+    ...G,
+    scores,
+    threat: {
+      ...G.threat,
+      modal: false,
     },
+    selectedThreat: G.threat.id,
+    identifiedThreats,
+  }
+}
+
+function draw(G, ctx, card) {
+  let deck = [...G.players[ctx.currentPlayer]];
```

Comment:
`const` please

Reply:
deck is changed with deck.splice (moves.js, line 199)
Does it still need to be a const

I thought you might have been talking about suit but

Reply:
Oh, if it gets changed, then maybe I'm just wrong.